iphone/Maps/Core/Theme/Renderers/UISearchBarRenderer.swift

```diff
@@ -86,10 +84,28 @@ class UISearchBarRenderer: UIViewRenderer {
         UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).font = font
       }
       if let fontColor = style.fontColor {
+        UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).defaultTextAttributes = [.foregroundColor: fontColor]
         UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).textColor = fontColor
         UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).leftView?.tintColor = fontColor
         UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).tintColor = fontColor
       }
     }
   }
+
+  // Draws the background image for the UITextField using the default system's text field height.
+  // This approach is used only for iOS 12.
+  private class func searchBarBackgroundImage(color: UIColor) -> UIImage? {
```

Comment:
1. Is the image exactly the same each time?
2. How often is it called?
3. Does it make sense to cache the returned image if it's the same and is called often, to save a bit of battery?

Reply:
1. This image is extended by the UITextField to fit size. But on iOS12 it doesn't works properly, so the size is needed.
2. I've take a look and found that this image built 2 times per appearing, so it is bad(( Thanks!
3. Continuing p2... I fix the code and chache current image into the static prop. So it is built once per app lifecycle. And is rebuilt only if appearance was changed (because other bg color is needed).